### PR TITLE
feat: t7 - get only the filenames

### PR DESCRIPTION
**Pull request by rtest12**
## Task 7 - get only the filenames

Script should get a folder as an argument and return a list of all unique files with extensions but without the absolute path (including in the nested directories). only -f option is allowed for the use of find command, i.e.: find /tmp -type f. Which should find all files in all nested dirs and then process the output by using expansion.

name the file: `chapter2-l2-t7-get-unique-files.sh`

example of the call:

`chapter2-l2-t7-get-unique-files.sh /tmp/bazel`
```
main.sh.x.c
main.c
dummy.c
zutil.c
uncompr.c
trees.c
minigzip.c
infcover.c
example.c
inftrees.c
```

## My output:
```
✗  ./chapter2-l2-t7-get-unique-files.sh /opt/Obsidian                          
af.pak
am.pak
app.asar
ar.pak
bg.pak
binding.node
bn.pak
ca.pak
...
```

### CODE REVIEW

File: chapter2/tests/t7/chapter2-l2-t7-get-unique-files.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+#
```
Comment (kseniyashaydurova) [inline chapter2/tests/t7/chapter2-l2-t7-get-unique-files.sh]:
```suggestion

```

File: chapter2/tests/t7/chapter2-l2-t7-get-unique-files.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+
+# Find only regular files and put in array.
+files=$(find "$1" -type f)
+
+
+# Leave only unique files, without extensions
+find "$1" -type f | while read file; do
+  echo "${file##*/}"
+done | sort -u
```
Comment (kseniyashaydurova) [inline chapter2/tests/t7/chapter2-l2-t7-get-unique-files.sh:26]: Please always add 1 empty line at the end

File: chapter2/tests/t7/chapter2-l2-t7-get-unique-files.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+
+# args check
+if [ "$#" -ne 1 ]; then
+  err 255 'Specify a single directory.'
+fi
+
+
+# Find only regular files and put in array.
+files=$(find "$1" -type f)
```
Comment (kseniyashaydurova) [inline chapter2/tests/t7/chapter2-l2-t7-get-unique-files.sh]: Is `files` var used? I don't see that. I see this logic repetition in 29th line

**Review by kseniyashaydurova — CHANGES_REQUESTED** [review]

File: chapter2/tests/t7/chapter2-l2-t7-get-unique-files.sh (hunk 2 above)
Comment (rtest12) [reply chapter2/tests/t7/chapter2-l2-t7-get-unique-files.sh:26]: But it exist.. Both in task 6 and in task 7.

File: chapter2/tests/t7/chapter2-l2-t7-get-unique-files.sh (hunk 3 above)
Comment (rtest12) [reply chapter2/tests/t7/chapter2-l2-t7-get-unique-files.sh]: > Is `files` var used? I don't see that. I see this logic repetition in 29th line

My mistake, as in the 6th task, I redid it but did not remove the excess. Fixed

**Review by kseniyashaydurova — APPROVED** [review]

**Pull request merged**